patroni/ha.py

```diff
@@ -867,6 +872,11 @@ def process_unhealthy_cluster(self):
 
         if self.is_healthiest_node():
             if self.acquire_lock():
+
+                if not self.call_pre_promote(self.patroni.config):
```

Comment:
In case is pre_romote call takes longer than `ttl` the leader lock will expire, therefore it must be executed with AsyncExecutor.